### PR TITLE
Issue 1064

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)
#1064 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced. 
Please also list any relevant details for your test configuration.

- [ ] Test A: sanity test
- [ ] Test B: regression test

- UT/IT execution results

- Logs
[regression test result](https://github.com/pytorch/serve/files/6717632/reg.txt)

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

Code changes description:
- model gen automation 
1. ts_scripts/marsgen.py
2. ts_scripts/mar_config.json

- sanity test
1. ts_scripts/sanity_utils.py
2. ts_scripts/tsutils.py

- regression test
1. ts_scripts/regression_utils.py 

- postman
1. test/postman/*

- grpc
1. ts_scripts/torchserve_grpc_client.py

- pytest
1. test/pytest/*

- doc
1. test/data_file_config.md

### CODE REVIEW

**Review by HamidShojanazeri — COMMENTED** [review]
Thanks @lxning , I could successfully test it, here are the regression [logs](https://gist.github.com/HamidShojanazeri/9281ff34091fd5647bd0e6ece4c528e6) for reference. Please refer to in-line comments.

File: test/data_file_config.md [hunk 1]
```diff
@@ -2,7 +2,7 @@ Use following properties to add inference test case in inference_data.json
 
 Mandatory properties
 ----
-**url:** Model url.
+**url:** Model url or a varialbe {{mar_path_xxxx}} defined in environment.json. xxxx is replaced with model name. This model's mar file is generated by ts_scripts/marsgen.py via ts_scripts/mar_config.json.
```
[the remark this answers is not in the document]
Comment (lxning) [reply test/data_file_config.md:5]: mar_path is the key used for identifying local mar or remote url. "mar_path_xxxx" is used in environment.json. Pls check.

File: test/postman/inference_data.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
         "tolerance":5
     },
     {
-        "url":"https://torchserve.pytorch.org/mar_files/squeezenet1_1.mar",
+        "url":"{{mar_path_squeezenet1_1}}",
```
[the remark this answers is not in the document]
Comment (lxning) [reply test/postman/inference_data.json:35]: text_classification has issue with PT1.9 so it is not handled at here.

File: ts_scripts/sanity_utils.py [hunk 3]
```diff
@@ -104,7 +102,10 @@ def test_sanity():
         model_handler = model["handler"]
 
         # Run gRPC sanity
-        register_model_grpc_cmd = f"python ts_scripts/torchserve_grpc_client.py register {model_name}"
+        print("pass mg.mar_set=", mg.mar_set)
```
Comment (nskool) [inline ts_scripts/sanity_utils.py:105]: Nit: this perhaps needs to be cleaned up

File: ts_scripts/torchserve_grpc_client.py [hunk 4]
```diff
@@ -33,9 +33,18 @@ def infer(stub, model_name, model_input):
         exit(1)
 
 
-def register(stub, model_name):
+def register(stub, model_name, mar_set_str):
+    mar_set = set()
+    if mar_set_str:
+        mar_set = set(mar_set_str.split(','))
+    marfile = f"{model_name}.mar"
+    print(f"## Check {marfile} in mar_set :", mar_set)
```
Comment (nskool) [inline ts_scripts/torchserve_grpc_client.py:41]: Nit: can be cleaned up

**Review by nskool — APPROVED** [review]

**Review by HamidShojanazeri — APPROVED** [review]

**Pull request merged**